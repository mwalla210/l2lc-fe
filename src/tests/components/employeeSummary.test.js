import React from 'react'
import renderer from 'react-test-renderer'
import EmployeeSummary from '../../components/employeeSummary'

jest.mock('../../components/barcode')
jest.mock('../../components/buttonDefault')
jest.mock('reactstrap', () => ({
  ButtonGroup: 'ButtonGroup',
  DropdownItem: 'DropdownItem',
  DropdownMenu: 'DropdownMenu',
  ButtonDropdown: 'ButtonDropdown',
  DropdownToggle: 'DropdownToggle',
}))

const defaultOptions = {
  page: {
    employeeEditPage: jest.fn(),
    changeCustomerPage: jest.fn(),
  },
  website: {
    currentEmployee: {
      fullName: 'fullName',
      barcodeDomID: 'barcodeDomID',
      barcodeScanID: 'barcodeScanID'
    },
    summaryActionsDropdownOpen: false,
    toggleSummaryActionsDD: jest.fn()
  }
}
describe('EmployeeSummary', () => {
  it ('Renders with snapshot (summaryActionsDropdownOpen: false)', () => {
    let options = Object.assign({}, defaultOptions)
    const component = renderer.create(
      <EmployeeSummary {...options}/>,
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it ('Renders with snapshot (summaryActionsDropdownOpen: true)', () => {
    let options = Object.assign({}, defaultOptions)
    options.website.summaryActionsDropdownOpen = true
    const component = renderer.create(
      <EmployeeSummary {...options}/>,
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it ('Renders and calls function printClick', () => {
    let options = Object.assign({}, defaultOptions)
    global.print = jest.fn()
    const component = renderer.create(
      <EmployeeSummary {...options}/>,
    )
    const inst = component.getInstance()
    inst.printClick()
    expect(global.print).toBeCalled()
  })
})
