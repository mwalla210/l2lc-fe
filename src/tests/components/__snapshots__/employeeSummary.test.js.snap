// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmployeeSummary Renders with snapshot (summaryActionsDropdownOpen: false) 1`] = `
<div>
  <div
    className="row justify-content-center"
  >
    <div
      className="col-3"
      style={
        Object {
          "border": "1px solid #b4b4b4",
          "borderRadius": "8px",
          "margin": "2px",
          "paddingTop": "1rem",
        }
      }
    >
      <div>
        <h6>
          Name
        </h6>
        <p>
          fullName
        </p>
      </div>
    </div>
  </div>
  <div
    className="row justify-content-center"
  />
  <div
    className="row justify-content-center"
  >
    <ButtonGroup>
      <ButtonDropdown
        isOpen={false}
        toggle={[MockFunction]}
      >
        <DropdownToggle
          caret={true}
          color="primary"
        >
          Actions
        </DropdownToggle>
        <DropdownMenu>
          <DropdownItem
            onClick={[MockFunction]}
          >
            Edit
          </DropdownItem>
        </DropdownMenu>
      </ButtonDropdown>
    </ButtonGroup>
  </div>
</div>
`;

exports[`EmployeeSummary Renders with snapshot (summaryActionsDropdownOpen: true) 1`] = `
<div>
  <div
    className="row justify-content-center"
  >
    <div
      className="col-3"
      style={
        Object {
          "border": "1px solid #b4b4b4",
          "borderRadius": "8px",
          "margin": "2px",
          "paddingTop": "1rem",
        }
      }
    >
      <div>
        <h6>
          Name
        </h6>
        <p>
          fullName
        </p>
      </div>
    </div>
  </div>
  <div
    className="row justify-content-center"
  />
  <div
    className="row justify-content-center"
  >
    <ButtonGroup>
      <ButtonDropdown
        isOpen={true}
        toggle={[MockFunction]}
      >
        <DropdownToggle
          caret={true}
          color="primary"
        >
          Actions
        </DropdownToggle>
        <DropdownMenu>
          <DropdownItem
            onClick={[MockFunction]}
          >
            Edit
          </DropdownItem>
        </DropdownMenu>
      </ButtonDropdown>
    </ButtonGroup>
  </div>
</div>
`;
